src/base/path.h

```diff
@@ -37,9 +37,7 @@
 
 #include "pathfwd.h"
 
-#include "base/interfaces/istringable.h"
-
-class Path final : public IStringable
+class Path final
```

Comment:
IMO this is the weak point. The class now doesn't indicate it satisfy some concept and the best workaround I could think of is to put `static_assert()` in .cpp file and this workaround isn't even a good one. And it wouldn't work for a template class.
The upside of using concepts is it has better expressions for constraints. For example, now it can check for specific constructor existence. This is something that the previous interface couldn't do.

I'm not pushing strongly for this PR. So unless there are better workarounds I might have to ditch this PR.